### DISCUSSION TITLE
Operands on HLIL_IF instructions

### DISCUSSION THREAD

**Discussion by Martyx00** [Vector35/binaryninja-api#2080] [API] [answered]
I am not sure if this is not intentional but `operands` (also `postfix` and `prefix` variants) is retrieved from an `HLIL_IF` instruction the returned list contains the operands for the body itself. 
So my question is whether there is a way of getting just the operands from the specific `HLIL_IF` instructions or if it can be added in one of future releases.
Example:
```
>>> ins
<HLIL_IF: if ($v0 != 0)>
>>> ins.postfix_operands
[<var int32_t $v0>, <HLIL_VAR 4>, 0, <HLIL_CONST 4>, <HLIL_CMP_NE>, [<HLIL_ASSIGN: $v0_1 = fscanf($v0, 0x53b1a8, &var_18)>, <HLIL_IF: if ($v0_1 == 1)>], <HLIL_BLOCK>, <HLIL_NOP>, <HLIL_IF>]
```

**Comment by psifertex** [comment, marked as answer]
Can you be more specific as to what output you want? There's a `.condition` and a `.condition.left` and a `.condition.right` property that might include what you want but I'm not sure whether I'm understanding the question correctly.
